tests/discovery.rs

```diff
@@ -328,6 +328,83 @@ macro_rules! generate_tests {
             assert_eq!(client.get("/"), "hello");
         }
 
+        mod remote_header {
+            use http::header::HeaderValue;
+            use super::super::*;
+
+            const REMOTE_IP_HEADER: &'static str = "l5d-remote-ip";
+            const FOO: &'static str = "0.0.0.0";
+            const BAR: &'static str = "127.0.0.1";
+
+            #[test]
+            fn outbound_should_strip() {
+                let _ = env_logger_init();
+                let header = HeaderValue::from_static(FOO);
+
+                let srv = $make_server().route_fn("/strip", |_req| {
+                    Response::builder().header(REMOTE_IP_HEADER, FOO).body(Default::default()).unwrap()
+                }).run();
+
+                let ctrl = controller::new().destination_and_close("disco.test.svc.cluster.local", srv.addr);
+                let proxy = proxy::new().controller(ctrl.run()).outbound(srv).run();
+                let client = $make_client(proxy.outbound, "disco.test.svc.cluster.local");
+                let rsp = client.request(&mut client.request_builder("/strip"));
```

Comment:
Nit/TIOLI: since you're not doing anything special with the request, I think this can just be
```suggestion
                let rsp = client.get("/strip");
```

Reply:
Since `.get` returns a `String`, it was either this, or change `Response::default()` to `builder()` and setting a body (in not wanting to just assert equality on `""`). I'll keep it how it is since changing it just introduces a small nit somewhere else.

Reply:
👍